src/pages/ListUsers.tsx

```diff
@@ -1,6 +1,70 @@
 import { H1 } from '@govuk-react/heading';
-import { ReactElement } from 'react';
+import { Fragment, ReactElement, useEffect, useState } from 'react';
+import { AttributeListType, UsersListType } from 'aws-sdk/clients/cognitoidentityserviceprovider';
+import Table from '@govuk-react/table';
+import { ATTRIBUTE_MAP, MAIN_USER_POOL_PREFIX, STATUS_MAP } from '../constants';
+import { getCognitoClient, getUserPoolList, listUsersInPool } from '../data/cognito';
 
-const ListUsers = (): ReactElement => <H1>User List</H1>;
+const formatAttributes = (attributes: AttributeListType) => {
+    return attributes
+        .filter((attribute) => ATTRIBUTE_MAP[attribute.Name])
+        .map((attribute) => (
+            <Fragment key={attribute.Name}>
+                <span>
+                    <strong>{ATTRIBUTE_MAP[attribute.Name]}</strong>: {attribute.Value}
+                </span>
+                <br />
+            </Fragment>
+        ));
+};
+
+const ListUsers = (): ReactElement => {
+    const [users, setUsers] = useState<UsersListType>([]);
+
+    useEffect(() => {
+        const getUsers = async (): Promise<UsersListType> => {
+            const cognito = await getCognitoClient();
+
+            const userPoolList = await getUserPoolList(cognito);
+            const mainUserPool = userPoolList?.find((pool) => pool.Name?.startsWith(MAIN_USER_POOL_PREFIX));
+
+            if (mainUserPool?.Id) {
+                return listUsersInPool(cognito, mainUserPool.Id);
+            }
+
+            console.error('Failed to retrieve main user pool data');
+
+            return [];
+        };
+
+        getUsers()
+            .then((data) => setUsers(data))
+            .catch((err) => {
+                console.error(err);
+
+                setUsers([]);
+            });
+    }, []);
```

Comment:
What's the purpose of this second argument passed to `useEffect`, `[]`?

Reply:
The second argument controls the hook dependencies, which are values that will trigger the hook whenever they change. So if we added the `users` variable to that array, the function would trigger every time `users` changed, which would probably lead to an error in this case since the function updates `users` itself.

So by passing an empty array, it's basically saying only run this on first render of the component.